### PR TITLE
KY-93637 |  Prometheus enable/disable

### PR DESCRIPTION
**Pull request by VJ-VijayThales**
Added suppor for prometheus enable disable 
Added support for getting status of prometheus

### CODE REVIEW

**Review by anugram — APPROVED** [review]
Looks good to me.

File: internal/provider/cm/resource_cm_prometheus.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+	tflog.Trace(ctx, common.MSG_METHOD_END+"[resource_cm_prometheus.go -> Enable/Disable]["+status+"]")
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *resourceCMPrometheus) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
```
Comment (om-shrirao) [inline internal/provider/cm/resource_cm_prometheus.go:101]: can we not add any message here?

File: internal/provider/cm/resource_cm_prometheus.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+func (r *resourceCMPrometheus) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+}
+
+func (r *resourceCMPrometheus) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	tflog.Trace(ctx, common.MSG_METHOD_START+"[resource_cm_prometheus.go -> Enable/Disable - Delete]")
+
+	var payload CMPrometheusMetricsConfigJSON
+	payload.Enabled = false
```
Comment (om-shrirao) [inline internal/provider/cm/resource_cm_prometheus.go:112]: did you confirm this with Nirav?

Comment (VJ-VijayThales) [reply internal/provider/cm/resource_cm_prometheus.go:112]: Yes, and decided to keep it false when destroying

**Review by om-shrirao — APPROVED** [review]

**Pull request merged**